transit_odp/pipelines/pipelines/dataset_etl/transform.py

```diff
@@ -78,6 +101,133 @@ def transform(self, extracted_data: ExtractedData) -> TransformedData:
             service_pattern_stops, service_patterns
         )
 
+        ### logic for flexible stop points transformation
+        # 1. extract naptan stoppoints data
+        if not flexible_stop_points.empty:
+            # 2. extract flexible zone data
+            flexible_zone = self.sync_flexible_zone(flexible_stop_points)
+            # 3. merge the flexible stop points and flexible zone to get the required geometry
+            flexible_stop_points_with_geometry = (
+                (
+                    flexible_stop_points.merge(
+                        flexible_zone,
+                        how="left",
+                        left_on="atco_code",
+                        right_on="atco_code",
+                    )
+                )
+                if not flexible_zone.empty
+                else flexible_stop_points
+            )
+            flexible_stop_points_with_geometry = transform_geometry(
```

Comment:
`transform_geometry` method seems redundant if you rename `flexible_location` field within `sync_flexible_zone` to `geometry`